### PR TITLE
Update test_precision_recall_curve.py

### PR DESCRIPTION
**Pull request by puhuk**
Description: Update for ddp config for updating in batch way

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/contrib/metrics/test_precision_recall_curve.py [hunk 1]
```diff
@@ -173,13 +171,22 @@ def _test(y_pred, y, batch_size, metric_device):
         assert precision_recall_curve(np_y, np_y_pred)[2] == pytest.approx(res[2].cpu().numpy())
 
     def get_test_cases():
+        torch.manual_seed(12 + rank)
         test_cases = [
             # Binary input data of shape (N,) or (N, 1)
-            (torch.randint(0, 2, size=(10,)), torch.randint(0, 2, size=(10,)), 1),
-            (torch.randint(0, 2, size=(10, 1)), torch.randint(0, 2, size=(10, 1)), 1),
+            (torch.randint(0, n_classes, size=(10,)), torch.randint(0, n_classes, size=(10,)), 1),
+            (torch.randint(0, n_classes, size=(10, 1)), torch.randint(0, n_classes, size=(10, 1)), 1),
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_precision_recall_curve.py:178]: IMO, there is no point to introduce `n_classes` for binary data, let's keep `randint(0, 2`

File: tests/ignite/contrib/metrics/test_precision_recall_curve.py [hunk 2]
```diff
@@ -138,20 +138,18 @@ def test_check_compute_fn():
 
 
 def _test_distrib_compute(device):
-
     rank = idist.get_rank()
-    torch.manual_seed(12)
+    n_iters = 80
+    batch_size = 16
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_precision_recall_curve.py:143]: I do not like this way to introduce some global constants that are partially used and even overridden, e.g. batch_size. Let's keep previous code: `n_iters = y.shape[0] // batch_size + 1` and `(torch.randint(0, 2, size=(50,)), torch.randint(0, 2, size=(50,)), 16),`

**Review by vfdev-5 — COMMENTED** [review]
Few nits to fix, otherwise lgtm. Thanks !

File: tests/ignite/contrib/metrics/test_precision_recall_curve.py (hunk 2 above)
Comment (puhuk) [reply tests/ignite/contrib/metrics/test_precision_recall_curve.py:143]: Let me correct and resend :)

**Review by vfdev-5 — DISMISSED** [review]
LGTM, thanks !

File: tests/ignite/contrib/metrics/test_precision_recall_curve.py [hunk 3]
```diff
@@ -173,6 +169,7 @@ def _test(y_pred, y, batch_size, metric_device):
         assert precision_recall_curve(np_y, np_y_pred)[2] == pytest.approx(res[2].cpu().numpy())
 
     def get_test_cases():
+        torch.manual_seed(12 + rank)
```
[the remark this answers is not in the document]
Comment (puhuk) [reply tests/ignite/contrib/metrics/test_precision_recall_curve.py:172]: @vfdev-5 I don't understand why outer loop does not work when seed is set here. with `get_test_cases`. It generates as expected. May I ask you to explain a little bit more.

[the remark this answers is not in the document]
Comment (vfdev-5) [reply tests/ignite/contrib/metrics/test_precision_recall_curve.py:172]: Outer loop works but we generate the **same** data 3 times which useless

[the remark this answers is not in the document]
Comment (puhuk) [reply tests/ignite/contrib/metrics/test_precision_recall_curve.py:172]: Oh, ok let me check again and correct it.

**Review by vfdev-5 — APPROVED** [review]
Thanks @puhuk !

**Pull request merged**